TylerJang27/trunk-action
# Upgrade mode for action

Adds a sub-action for auto-generating a trunk PR with the result of `trunk upgrade`. Enabling this behavior requires adding the following to a nightly (or weekly cron workflow):

```yaml
name: Nightly
on:
  schedule:
    - cron: 0 8 * * 1-5
  workflow_dispatch: {}

permissions: read-all

jobs:
  trunk_upgrade:
    name: Upgrade Trunk
    runs-on: ubuntu-latest
    permissions:
      contents: write # For trunk to create PRs
      pull-requests: write # For trunk to create PRs

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      # >>> Install your own deps here (npm install, etc) <<<

      - name: Trunk Upgrade
        uses: trunk-io/trunk-action/upgrade@v1
```

Manual testing I conducted:
1. Running with [packages](https://github.com/TylerJang27/trunk-action/pull/8)
2. Running with [different args/scopes](https://github.com/TylerJang27/trunk-action/pull/9) and [job](https://github.com/TylerJang27/trunk-action/actions/runs/4434987105/jobs/7781575163?pr=1)
3. Running with [bad argument](https://github.com/TylerJang27/trunk-action/actions/runs/4434974092/jobs/7781545650?pr=1) ("The following argument was not expected: runtimes")
4. Running with upgrade errors (did not test, since the conditions are difficult to simulate)
5. Running with [no .trunk/trunk.yaml](https://github.com/TylerJang27/trunk-action/actions/runs/4435018284/jobs/7781642909?pr=1)
6. Running from [nightly](https://github.com/TylerJang27/trunk-action/actions/runs/4435442084)
7. Running with [empty trunk upgrade result](https://github.com/TylerJang27/trunk-action/actions/runs/4434945939/jobs/7781483722?pr=1) ("Branch 'trunk-io/update-trunk-test-1' is not ahead of base 'main' and will not be created")
8. [Multi-OS](https://github.com/TylerJang27/trunk-action/pull/10) and [job](https://github.com/TylerJang27/trunk-action/actions/runs/4435037028/jobs/7781685491?pr=1)

Underwent some design review here (TODO: ADD LINK)
